### PR TITLE
Fix _int_mm CPU corner case

### LINKED ISSUE #195066: [cpu][Inductor] torch._int_mm silent failure on CPU when an input's size-1 dim has stride 0
### 🐛 Describe the bug

`torch._int_mm` on CPU silently failed when an input has a size-1 dimension with stride=0. The stride of a size-1 dim is unconstrained and stride=0 could be produced by inductor. This was found when testing some quantized LLM workload. Inductor emits stride-0 `(1, K)` reshape of a single-token activation, and torchao's `Int8DynamicActivationInt8WeightConfig` on CPU yields garbage output.

```python
import torch

a = torch.randint(-128, 127, (1, 512), dtype=torch.int8)
b = torch.randint(-128, 127, (512, 512), dtype=torch.int8)
ref = a.float() @ b.float()

# the stride of a size-1 dim is arbitrary; Inductor emits 0 for it when it
# reshapes a (1, 1, K) single-token activation to (1, K)
a0 = a.as_strided((1, 512), (0, 1))

out = torch.full((1, 512), -1, dtype=torch.int32)
torch._int_mm(a0, b, out=out)
print("output slots left untouched:", int((out == -1).sum()), "/", out.numel())
print("max abs error:", (out.float() - ref).abs().max().item())

with torch.backends.mkldnn.flags(enabled=False):
    out2 = torch._int_mm(a0, b)
print("max abs error with mkldnn disabled:", (out2.float() - ref).abs().max().item())
```

Output on torch-2.15.0.dev20260827+cpu:

```
output slots left untouched: 512 / 512
max abs error: 340278.0
max abs error with mkldnn disabled: 0.0
```

Expected: the correct output as the non-oneDNN path.

`_mkldnn_gemm_i8i8i32_with_blas` in `aten/src/ATen/native/mkldnn/Matmul.cpp` takes oneDNN's `lda`/`ldb` straight from the input strides. oneDNN's gemm requires `lda >= K` when `transa == 'N'`, so the call fails with `invalid_arguments` and the status returned is discarded. Seems `_int_mm_out_cpu` records the call as dispatched and skips its own reference gemm. I already have a darft fix https://github.com/pytorch/pytorch/pull/194682.

### Versions

PyTorch version: 2.15.0.dev20260827+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.4 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 4.4.2
Libc version: glibc-2.39

Python version: 3.10.19 | packaged by conda-forge | (main, Oct 22 2025, 22:29:10) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-137-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  288
On-line CPU(s) list:                     0-287
Vendor ID:                               GenuineIntel
Model name:                              Intel(R) Xeon(R) 6960P
CPU family:                              6
Model:                                   173
Thread(s) per core:                      2
Core(s) per socket:                      72
Socket(s):                               2
Stepping:                                1
CPU(s) scaling MHz:                      21%
CPU max MHz:                             3900.0000
CPU min MHz:                             800.0000
BogoMIPS:                                5400.00
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect user_shstk avx_vnni avx512_bf16 wbnoinvd dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req hfi vnmi avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr ibt amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities ibpb_exit_to_user
Virtualization:                          VT-x
L1d cache:                               6.8 MiB (144 instances)
L1i cache:                               9 MiB (144 instances)
L2 cache:                                288 MiB (144 instances)
L3 cache:                                864 MiB (2 instances)
NUMA node(s):                            2
NUMA node0 CPU(s):                       0-71,144-215
NUMA node1 CPU(s):                       72-143,216-287
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS Not affected; BHI BHI_DIS_S
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Mitigation; IBPB before exit to userspace

Versions of relevant libraries:
[pip3] intel-cmplr-lib-ur==2026.1.1
[pip3] intel-openmp==2026.1.1
[pip3] mkl-include==2026.1.0
[pip3] mkl-static==2026.1.0
[pip3] numpy==2.2.6
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvtx==13.0.85
[pip3] oneccl-bind-pt==2.2.0+cpu
[pip3] onemkl-license==2026.1.0
[pip3] optree==0.19.1
[pip3] tbb==2023.1.0
[pip3] tbb-devel==2023.1.0
[pip3] tcmlib==1.5.0
[pip3] torch==2.15.0.dev20260827+cpu
[pip3] torchao==0.19.0.dev20260826+cpu
[pip3] torchaudio==2.11.0
[pip3] torchvision==0.30.0.dev20260825+cpu
[pip3] triton==3.8.0+gitdf3f91dd
[pip3] umf==1.1.0
[conda] intel-cmplr-lib-ur           2026.1.1                 pypi_0                pypi
[conda] intel-openmp                 2026.1.1                 pypi_0                pypi
[conda] mkl-include                  2026.1.0                 pypi_0                pypi
[conda] mkl-static                   2026.1.0                 pypi_0                pypi
[conda] numpy                        2.2.6                    pypi_0                pypi
[conda] nvidia-cublas                13.1.1.3                 pypi_0                pypi
[conda] nvidia-cuda-cupti            13.0.85                  pypi_0                pypi
[conda] nvidia-cuda-nvrtc            13.0.88                  pypi_0                pypi
[conda] nvidia-cuda-runtime          13.0.96                  pypi_0                pypi
[conda] nvidia-cudnn-cu13            9.20.0.48                pypi_0                pypi
[conda] nvidia-cufft                 12.0.0.61                pypi_0                pypi
[conda] nvidia-curand                10.4.0.35                pypi_0                pypi
[conda] nvidia-cusolver              12.0.4.66                pypi_0                pypi
[conda] nvidia-cusparse              12.6.3.3                 pypi_0                pypi
[conda] nvidia-cusparselt-cu13       0.8.1                    pypi_0                pypi
[conda] nvidia-nccl-cu13             2.29.7                   pypi_0                pypi
[conda] nvidia-nvjitlink             13.0.88                  pypi_0                pypi
[conda] nvidia-nvtx                  13.0.85                  pypi_0                pypi
[conda] onemkl-license               2026.1.0                 pypi_0                pypi
[conda] optree                       0.19.1                   pypi_0                pypi
[conda] tbb                          2023.1.0                 pypi_0                pypi
[conda] tbb-devel                    2023.1.0                 pypi_0                pypi
[conda] tcmlib                       1.5.0                    pypi_0                pypi
[conda] torch                        2.15.0.dev20260827+cpu   pypi_0                pypi
[conda] torchao                      0.19.0.dev20260826+cpu   pypi_0                pypi
[conda] torchaudio                   2.11.0                   pypi_0                pypi
[conda] torchvision                  0.30.0.dev20260825+cpu   pypi_0                pypi
[conda] triton                       3.8.0+gitdf3f91dd        pypi_0                pypi
[conda] umf                          1.1.0                    pypi_0                pypi

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen @snadampal

### PR DESCRIPTION
**Pull request by Kanya-Mo**
This is to fix https://github.com/pytorch/pytorch/issues/195066 _mkldnn_gemm_i8i8i32_with_blas took leading dimensions from the input strides, but the stride of a size-1 dim is arbitrary and inductor emits 0 for it in some cases. On the other hand, oneDNN's row-major gemm requires lda >= K so it will silently reject the call, causing accuracy failures in workloads like int8 quantized LLM decoding.

The PR fixes this corner case and also check this oneDNN gemm status so that exception can be raised. Authored with the assistance of Claude Code.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01

### CODE REVIEW

File: aten/src/ATen/native/mkldnn/Matmul.cpp
Comment (frost-intel) [inline aten/src/ATen/native/mkldnn/Matmul.cpp:624]: If a concern is that the stride can be 0 or arbitrary, this means actually contiguous inputs would not be marked as such here, so can we improve this to identify the calls that could be going to the blas API but currently aren't? I'd expect a significant perf improvement.

File: aten/src/ATen/native/mkldnn/Matmul.cpp [hunk 1]
```diff
@@ -596,11 +596,17 @@ static void _mkldnn_gemm_i8i8i32_with_blas(
...
+      status = CALL_DNNL_GEMM(dnnl::gemm_u8s8s32, uint8_t*, self.data_ptr());
     } else { //int8
-      CALL_DNNL_GEMM(dnnl::gemm_s8s8s32, int8_t*, self.data_ptr());
+      status = CALL_DNNL_GEMM(dnnl::gemm_s8s8s32, int8_t*, self.data_ptr());
     }
+
+    TORCH_CHECK(
+      status == dnnl::status::success,
+      "oneDNN i8i8i32 gemm failed with status ",
+      static_cast<int>(status));
```
Comment (frost-intel) [inline aten/src/ATen/native/mkldnn/Matmul.cpp]: This message isn't really actionable. Consider reporting transa, transb, m, n, etc?

**Review by frost-intel — CHANGES_REQUESTED** [review]
You should add a test, probably in test/test_linalg.py::test__int_mm_cpu.

File: aten/src/ATen/native/mkldnn/Matmul.cpp
Comment (Kanya-Mo) [reply aten/src/ATen/native/mkldnn/Matmul.cpp:624]: Yes, that should be such cases. Working on this as a follow up PR.

**Review by frost-intel — APPROVED** [review]

**Comment by frost-intel** [comment]
@aditew01 or @Skylion007  Could you review this? It's not directly conflicting with https://github.com/pytorch/pytorch/pull/180455 but touching Matmul.cpp

**Review by aditew01 — APPROVED** [review]
thanks for pushing the fix. @Skylion007  can you please take a look ?

**Comment by frost-intel** [comment]
@Skylion007 Could you please take a look? Thanks.
